src/sinks/prometheus/exporter.rs

```diff
@@ -515,13 +511,16 @@ impl PrometheusExporter {
         });
 
         self.server_shutdown_trigger = Some(trigger);
+        Ok(())
     }
 }
 
 #[async_trait]
 impl StreamSink<Event> for PrometheusExporter {
     async fn run(mut self: Box<Self>, mut input: BoxStream<'_, Event>) -> Result<(), ()> {
-        self.start_server_if_needed().await;
+        self.start_server_if_needed()
+            .await
+            .map_err(|error| error!("Start server error: {}.", error))?;
```

Comment:
This error message should be something more appropriate, like `Failed to start Prometheus exporter: {}.`.

Reply:
Fixed